tc/sgx/trusted_worker_manager/enclave/signup_enclave_common.h

```diff
@@ -0,0 +1,28 @@
+/* Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include "tcf_error.h"
+
+tcf_err_t ecall_CalculateSealedEnclaveDataSize(size_t* pSealedEnclaveDataSize);
+
+tcf_err_t ecall_CalculatePublicEnclaveDataSize(size_t* pPublicEnclaveDataSize);
+
+tcf_err_t ecall_CreateEnclaveData(size_t* outPublicEnclaveDataSize,
+    size_t* outSealedEnclaveDataSize);
+
+tcf_err_t ecall_UnsealEnclaveData(char* outPublicEnclaveData,
+    size_t inAllocatedPublicEnclaveDataSize);
```

Comment:
Not sure why is this called common. Sealing is not applicable for WPE. Do you mean common for Singleton and KME. May be call it signup_seal.h ?

Creating C++ class might be better option than functional coding approach. But if changes are too much to do in this sprint, that can be done later.

Reply:
I named this file as signup_enclave_common because all functions except UnsealEnclaveData are common. the function UnsealEnclaveData can be stopped from calling even at the untrusted layer itself by not providing the wrapper function for the same function (for WPE).

I thought about creating class in trusted code, but i am not sure about any way to instantiate class in trusted layer from untrusted and call actual ecalls.